go.mod

```
@@ -4,75 +4,72 @@ go 1.15
+	k8s.io/component-base => github.com/k3s-io/kubernetes/staging/src/k8s.io/component-base v1.21.0-k3s1
+	k8s.io/component-helpers => github.com/k3s-io/kubernetes/staging/src/k8s.io/component-helpers v1.21.0-k3s1
+	k8s.io/controller-manager => github.com/k3s-io/kubernetes/staging/src/k8s.io/controller-manager v1.21.0-k3s1
+	k8s.io/cri-api => github.com/k3s-io/kubernetes/staging/src/k8s.io/cri-api v1.21.0-k3s1
+	k8s.io/csi-translation-lib => github.com/k3s-io/kubernetes/staging/src/k8s.io/csi-translation-lib v1.21.0-k3s1
+	k8s.io/kube-aggregator => github.com/k3s-io/kubernetes/staging/src/k8s.io/kube-aggregator v1.21.0-k3s1
+	k8s.io/kube-controller-manager => github.com/k3s-io/kubernetes/staging/src/k8s.io/kube-controller-manager v1.21.0-k3s1
+	k8s.io/kube-proxy => github.com/k3s-io/kubernetes/staging/src/k8s.io/kube-proxy v1.21.0-k3s1
+	k8s.io/kube-scheduler => github.com/k3s-io/kubernetes/staging/src/k8s.io/kube-scheduler v1.21.0-k3s1
+	k8s.io/kubectl => github.com/k3s-io/kubernetes/staging/src/k8s.io/kubectl v1.21.0-k3s1
+	k8s.io/kubelet => github.com/k3s-io/kubernetes/staging/src/k8s.io/kubelet v1.21.0-k3s1
+	k8s.io/kubernetes => github.com/k3s-io/kubernetes v1.21.0-k3s1
+	k8s.io/legacy-cloud-providers => github.com/k3s-io/kubernetes/staging/src/k8s.io/legacy-cloud-providers v1.21.0-k3s1
+	k8s.io/metrics => github.com/k3s-io/kubernetes/staging/src/k8s.io/metrics v1.21.0-k3s1
+	k8s.io/mount-utils => github.com/k3s-io/kubernetes/staging/src/k8s.io/mount-utils v1.21.0-k3s1
+	k8s.io/node-api => github.com/k3s-io/kubernetes/staging/src/k8s.io/node-api v1.21.0-k3s1
+	k8s.io/sample-apiserver => github.com/k3s-io/kubernetes/staging/src/k8s.io/sample-apiserver v1.21.0-k3s1
+	k8s.io/sample-cli-plugin => github.com/k3s-io/kubernetes/staging/src/k8s.io/sample-cli-plugin v1.21.0-k3s1
+	k8s.io/sample-controller => github.com/k3s-io/kubernetes/staging/src/k8s.io/sample-controller v1.21.0-k3s1
+	mvdan.cc/unparam => mvdan.cc/unparam v0.0.0-20210104141923-aac4ce9116a7
 )
 
 require (
 	github.com/containerd/continuity v0.0.0-20210208174643-50096c924a4e
-	github.com/frankban/quicktest v1.11.3 // indirect
-	github.com/google/go-containerregistry v0.4.2-0.20210316173552-70c58c0e4786
-	github.com/k3s-io/helm-controller v0.8.4
-	github.com/klauspost/compress v1.11.7
-	github.com/pierrec/lz4 v2.5.2+incompatible
+	github.com/google/go-containerregistry v0.5.0
+	github.com/k3s-io/helm-controller v0.9.1
 	github.com/pkg/errors v0.9.1
-	github.com/rancher/k3s v1.21.0-alpha1.0.20210416213554-c2eff9298ca7
+	github.com/rancher/k3s v1.21.1-0.20210513091809-079620ded0c0
+	github.com/rancher/wharfie v0.3.5
 	github.com/rancher/wrangler v0.6.2
 	github.com/rancher/wrangler-api v0.6.0
 	github.com/sirupsen/logrus v1.8.1
 	github.com/urfave/cli v1.22.2
 	google.golang.org/grpc v1.37.0
-	gopkg.in/yaml.v2 v2.4.0
-	k8s.io/api v1.20.5
-	k8s.io/apimachinery v1.20.5
-	k8s.io/apiserver v1.20.0
+	k8s.io/api v1.21.1
+	k8s.io/apimachinery v1.21.1
+	k8s.io/apiserver v1.21.1
```

These need to be the kubernetes version, with the `v1` replaced with `v0`. Its not super important since they're being replaced up above, but this is how Kubernetes versions the API modules.
```suggestion
	k8s.io/api v0.21.0
	k8s.io/apimachinery v0.21.0
	k8s.io/apiserver v0.21.0
```